Project-Hotpot/Project-Hotpot/Algorithm/SongAlgorithm.swift

```diff
@@ -41,9 +41,9 @@ class SongAlgorithm{
     }
     
     func getRandomGenre() -> String? {
-        //returns nil if the queue is empty
+        //returns nil if the queue somehow returns a nil object, which cannot happen
```

Comment:
Hmm... why does the signature of `GenreManager`'s `getGenre` return an optional if that value cannot be `nil`?

Non-optional variables are very powerful, they allow you to rely on having a value without requiring unnecessary nil checks to guard against NPEs.

Reply:
(note, this is also why I dislike `!` so much: it breaks this contract and robs non-optional variables of their power to abolish unnecessary NPE checks. Thank you for steering clear of `!`!)